### PR TITLE
All in a day's work

### LINKED ISSUE #420: 🛠 Adding step extensions breaks the editor a bit
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
When trying to add a new step to an integration after I've uploaded some extensions to check the icons:

image

### PR DESCRIPTION
**Pull request by gashcrumb**
Since I broke the editor by uploading some extensions 😹 

fixes #420
fixes syndesisio/syndesis#5604
fixes syndesisio/syndesis#5612
fixes syndesisio/syndesis#5255

Now I've gone and hidden away all the icon functions and created `EntityIcon` to display icons properly.  With this PR icons work nicely:

image

the openshift icon from that extension looks a little gross in this page, I don't know if the png file that was used in that extension maybe isn't square or what.  Also the backend doesn't appear to cache extension icons, so you'll see it loads a bunch slower than everything else but I can't do anything about that.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2200/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
Also I might have snuck in a fun "unknown" icon to kick people in the seat to provide icons :-)

File: app/ui-react/packages/ui/src/Connection/ConnectionCard.tsx [hunk 1]
```diff
@@ -242,7 +242,7 @@ export class ConnectionCard extends React.PureComponent<
             <Card.Body>
               <div className={'connection-card__body'}>
                 <div className="connection-card__icon">
-                  <img src={this.props.icon} alt={this.props.name} width={46} />
+                  {this.props.icon && this.props.icon}
```
Comment (riccardo-forina) [inline app/ui-react/packages/ui/src/Connection/ConnectionCard.tsx]: Is this && needed?

File: app/ui-react/packages/ui/src/Integration/IntegrationStepsHorizontalItem.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export class IntegrationStepsHorizontalItem extends React.Component<
         {!this.props.href && (
           <div>
             <div className={'step-icon'} title={this.props.title}>
-              <img src={this.props.icon} />
+              {this.props.icon && this.props.icon}
```
Comment (riccardo-forina) [inline app/ui-react/packages/ui/src/Integration/IntegrationStepsHorizontalItem.tsx]: Same here

File: app/ui-react/packages/ui/src/Integration/IntegrationStepsHorizontalItem.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ export class IntegrationStepsHorizontalItem extends React.Component<
           <Link to={this.props.href}>
             <div>
               <div className={'step-icon'} title={this.props.title}>
-                <img src={this.props.icon} />
+                {this.props.icon && this.props.icon}
```
Comment (riccardo-forina) [inline app/ui-react/packages/ui/src/Integration/IntegrationStepsHorizontalItem.tsx]: And here

File: app/ui-react/syndesis/public/icons/unknown.svg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (riccardo-forina) [inline app/ui-react/syndesis/public/icons/unknown.svg]: 😂

File: app/ui-react/syndesis/src/modules/integrations/components/editor/step/WithConfigurationForm.tsx [hunk 5]
```diff
@@ -108,7 +108,7 @@ export const WithConfigurationForm: React.FunctionComponent<
     definition = step.properties;
   }
 
-  const initialValue = step.configuredProperties;
+  const initialValue = props.step.configuredProperties;
```
Comment (riccardo-forina) [inline app/ui-react/syndesis/src/modules/integrations/components/editor/step/WithConfigurationForm.tsx]: 🤦‍♂️ sorry

**Comment by gashcrumb** [comment]
@paoloantinori I think I spotted that Nicola had put these icons already in the react UI in the main repo, so heads up they could conflict.

File: app/ui-react/packages/ui/src/Connection/ConnectionCard.tsx (hunk 1 above)
Comment (gashcrumb) [reply app/ui-react/packages/ui/src/Connection/ConnectionCard.tsx]: lemme check if I changed that to be required or not.

File: app/ui-react/syndesis/src/modules/integrations/components/editor/step/WithConfigurationForm.tsx (hunk 5 above)
Comment (gashcrumb) [reply app/ui-react/syndesis/src/modules/integrations/components/editor/step/WithConfigurationForm.tsx]: I knew you'd hate to see that, no worries though!  at least it was something simple :-)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2220/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Pull request merged**